agent/app/agent.go

```diff
@@ -730,8 +731,7 @@ func (agent *ecsAgent) startACSSession(
 		seelog.Criticalf("Unretriable error starting communicating with ACS: %v", err)
 		return exitcodes.ExitTerminal
 	}
-	seelog.Critical("ACS Session handler should never exit")
-	return exitcodes.ExitError
+	return exitcodes.ExitSuccess
```

Comment:
if this was classified as unexpected exit, why are we changing the exit as success? 
init has logic to handle agent stops based on the exit codes. Could you verify if this change does not affect init's behavior? same for terminal handler below.

Reply:
since we are now cancelling the context, we expect ACS to exit when we receive a termination signal, so this is no longer an error situation. I have verified this doesn't affect init's behavior.

Reply:
Is it true that this case will never be hit anymore?
https://github.com/aws/amazon-ecs-init/blob/d0ee8ae2e02a0e17f1d248b6b85bb1ff808e75d3/ecs-init/engine/engine.go#L231
If yes, lets change this

Reply:
There are a few cases where we can still exit terminal, such as when the ACS session returns an error or if saving state fails.

Reply:
cool, I see that terminal handler for state save failure is changed back to exit with error code.

Reply:
yes, removing that case was a mistake, it would be nice to have os.Exit handled in one place but would be a much larger refactor.